### PR TITLE
feat: add scheduled compact revision persistent

### PR DESCRIPTION
**Pull request by themanforfree**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
add scheduled compact revision persistent
* what changes does this pull request make?
add scheduled compact revision persistent
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
no

### CODE REVIEW

File: crates/xline/src/storage/db.rs [hunk 1]
```diff
@@ -58,6 +60,32 @@ impl DB {
             engine: Arc::new(engine),
         }))
     }
+
+    /// Get del lease key buffer
+    fn get_del_lease_key_buffer(ops: &[WriteOp]) -> HashMap<i64, Vec<u8>> {
```
Comment (Phoenix500526) [inline crates/xline/src/storage/db.rs:65]: Better to inline this function

File: crates/xline/src/storage/db.rs [hunk 2]
```diff
@@ -58,6 +60,32 @@ impl DB {
...
+                    Some((lease_id, lease_id.encode_to_vec()))
+                } else {
+                    None
+                }
+            })
+            .collect::<HashMap<_, _>>()
+    }
+
+    /// get del alarm buffer
+    fn get_del_alarm_buffer(ops: &[WriteOp]) -> Vec<u8> {
```
Comment (Phoenix500526) [inline crates/xline/src/storage/db.rs:78]: Same as above

Comment (themanforfree) [reply crates/xline/src/storage/db.rs:80]: Added

File: crates/xline/src/storage/db.rs (hunk 1 above)
Comment (themanforfree) [reply crates/xline/src/storage/db.rs:66]: Added

**Review by Phoenix500526 — APPROVED** [review]
LGTM

File: crates/xline/src/storage/db.rs [hunk 3]
```diff
@@ -50,6 +52,34 @@ impl DB {
             engine: Arc::new(engine),
         }))
     }
+
+    /// Get del lease key buffer
+    #[inline]
```
Comment (iGxnon) [inline crates/xline/src/storage/db.rs:57]: Private function maybe always inlined by llvm.
> Second, it usually isn’t necessary to apply #[inline] to private functions — within a crate, the compiler generally makes good inline decisions. There’s [a joke](https://twitter.com/ManishEarth/status/936084757212946432) that LLVM’s heuristic for when the function should be inlined is “yes”.
> —— https://matklad.github.io/2021/07/09/inline-in-rust.html

**Review by iGxnon — APPROVED** [review]

**Pull request merged**